common/src/indexer_errors.rs

```diff
@@ -1,353 +0,0 @@
-            Self::IE061 => "Failed to allocate: Invalid allocation amount provided",
-            Self::IE062 => "Did not receive tx receipt, not authorized or network paused",
-            Self::IE063 => "No active allocation with provided id found",
-            Self::IE064 => {
-                "Failed to unallocate: Allocation cannot be closed in the same epoch it was created"
-            }
-            Self::IE065 => "Failed to unallocate: Allocation has already been closed",
-            Self::IE066 => "Failed to allocate: allocation ID already exists on chain",
-            Self::IE067 => "Failed to query POI for current epoch start block",
-            Self::IE068 => "User-provided POI did not match reference POI from graph-node",
-            Self::IE069 => "Failed to query Epoch Block Oracle Subgraph",
-            Self::IE070 => "Failed to query latest valid epoch and block hash",
-            Self::IE071 => "Add Epoch subgraph support for non-protocol chains",
-            Self::IE072 => "Failed to execute batch tx (contract: staking)",
-            Self::IE073 => "Failed to query subgraph features from indexing statuses endpoint",
-            Self::IE074 => "Failed to resolve the release version",
-            Self::IE075 => "Failed to parse response body to query string",
-        }
-    }
-
-    pub fn explanation(&self) -> String {
-        format!("{}#{}", ERROR_BASE_URL, self.message())
-    }
-}
-
-#[derive(Debug)]
-pub struct IndexerErrorCause(Box<dyn Error + Send + Sync>);
-
-impl IndexerErrorCause {
-    pub fn new<E>(error: E) -> Self
-    where
-        E: Into<Box<dyn Error + Send + Sync>>,
-    {
-        IndexerErrorCause(error.into())
-    }
-}
-
-impl Display for IndexerErrorCause {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.0.fmt(f)
-    }
-}
-
-impl Error for IndexerErrorCause {
-    fn source(&self) -> Option<&(dyn Error + 'static)> {
-        self.0.source()
-    }
-}
-
-impl From<String> for IndexerErrorCause {
-    fn from(error: String) -> Self {
-        Self(Box::new(std::io::Error::new(
-            std::io::ErrorKind::Other,
-            error,
-        )))
-    }
-}
-
-#[derive(Debug)]
-pub struct IndexerError {
-    code: IndexerErrorCode,
-    explanation: String,
-    cause: Option<IndexerErrorCause>,
-}
-
-impl IndexerError {
-    // Create Indexer Error and automatically increment counter by the error code
-    pub fn new(code: IndexerErrorCode, cause: Option<IndexerErrorCause>) -> Self {
-        metrics::INDEXER_ERROR
-            .with_label_values(&[&code.to_string()])
-            .inc();
-        let explanation = code.message();
-        warn!(
-            "Encountered error {}: {}. Cause: {:#?}",
-            code.to_string(),
-            explanation,
-            cause
-        );
-        Self {
-            code,
-            explanation: explanation.to_string(),
-            cause,
-        }
```

Comment:
Some indexers would like to have the error information on their metrics. This means bringing back error codes mapping. This could be on a following PR depending on priorities for having this kind of metric.